scripts/TrapSpike.gd

```diff
@@ -0,0 +1,44 @@
+extends Area2D
+
+export var time_initial_delay = 0.0
+export var time_disabled = 2.0
+export var time_fadein = 0.5
+export var time_enabled = 0.5
+
+signal trap_triggered
+
+enum TrapSpikeState {
+    Disabled,
+    Fadein,
+    Enabled,
+}
+
+var _state = TrapSpikeState.Disabled
+
+func _ready():
+    add_to_group("traps")
+    _start_trap_cycle()
+
+func _physics_process(delta):
+    if _state == TrapSpikeState.Enabled:
+        for body in self.get_overlapping_bodies():
+            if body.name == "Player":
+                emit_signal("trap_triggered")
```

Comment:
Just FYI, maybe we'll be able to get rid of `if body.name == "Player"` after we implement collision layers and masks.